cmd/oras/manifest/fetch.go

```diff
@@ -72,46 +79,88 @@ Example - Fetch manifest with prettified json result:
 		},
 	}
 
-	cmd.Flags().BoolVarP(&opts.pretty, "pretty", "", false, "output prettified manifest")
-	cmd.Flags().BoolVarP(&opts.fetchDescriptor, "descriptor", "", false, "fetch a descriptor of the manifest")
 	cmd.Flags().StringSliceVarP(&opts.mediaTypes, "media-type", "", nil, "accepted media types")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
 	option.ApplyFlags(&opts, cmd.Flags())
 	return cmd
 }
 
 func fetchManifest(opts fetchOptions) error {
 	ctx, _ := opts.SetLoggerLevel()
-	targetPlatform, err := opts.Parse()
-	if err != nil {
-		return err
-	}
+
 	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
 	if err != nil {
 		return err
 	}
+
 	if repo.Reference.Reference == "" {
 		return oerrors.NewErrInvalidReference(repo.Reference)
 	}
 	repo.ManifestMediaTypes = opts.mediaTypes
 
-	// Fetch and output
-	var content []byte
-	if opts.fetchDescriptor {
-		content, err = cas.FetchDescriptor(ctx, repo, opts.targetRef, targetPlatform)
-	} else {
-		content, err = cas.FetchManifest(ctx, repo, opts.targetRef, targetPlatform)
+	targetPlatform, err := opts.Parse()
+	if err != nil {
+		return err
+	}
+
+	var src oras.ReadOnlyTarget = repo.Manifests()
+	if opts.cacheRoot != "" {
+		ociStore, err := oci.New(opts.cacheRoot)
+		if err != nil {
+			return err
+		}
+		src = cache.New(src, ociStore)
 	}
+
+	// fetch manifest
+	desc, content, err := oras.FetchBytes(ctx, src, opts.targetRef, oras.FetchBytesOptions{
```

Comment:
What if only `descriptor` is needed?
For instance, `oras manifest fetch --descriptor localhost:5000/hello:latest`?

Reply:
If only descriptor is needed, we don't need to fetch the content, which can save a lot of effort and improve the performance. I'll make a change in this way.

For `oras manifest fetch --descriptor localhost:5000/hello:latest`, only descriptor will be outputted to stdout, line 128 to 138.